### ISSUE TITLE
Extend AddConnectivity to support summing over multiple SRC_ID

### ISSUE THREAD

**Issue by tclune** [GEOS-ESM/MAPL#484]
"Anton, Elliott:
 
   It appears that we may need to provide some mechanism to control this 1-way coupling. Some thoughts,
 
For sure, this coupling must only involve ACTIVE instances as PASSIVE instances are meant to be “invisible” (does not impact anything else, by definition)
Now, among the active DU, SS and SU components, as long as they are ACTIVE, they should influence Nitrates.
The options now are to either pass a bundle and let nitrates do the summation, or else extend MAPL to that it performs the sum. For example, a typical add connectivity (in a different context) reads:
 
CALL MAPL_AddConnectivity ( GC, &
                   SRC_NAME  = (/ 'du001', 'du002', 'du003', 'du004', 'du005’/),
                   DST_NAME  = (/ 'du001', 'du002', 'du003', 'du004', 'du005 /), &
                   DST_ID = NI, SRC_ID = DU, __RC__  )
 
where the SRC_NAME and DST_NAME do not have to be the same (when they are as in this example, you can omit DST_NAME). Overloading MAPL so that SRC_ID could be a list of components
 
CALL MAPL_AddConnectivity ( GC, &
                   SRC_NAME  = (/ 'du001', 'du002', 'du003', 'du004', 'du005’/),
                   DST_NAME  = (/ 'du001', 'du002', 'du003', 'du004', 'du005 /), &
                   DST_ID = NI, SRC_ID = (/ DU.mineral, DU.ash /), __RC__  )
 
would have MAPL adding these all up for you. Sure, Atanas will need to support this and I have no idea how hard this would be – that is why I am cc’ing him. Atanas, what do you say? Preparing bundles and having nitrates do the summation as Anton suggests works as well, but you will need to write the support code for it. 
 
Now, it gets more complicated. What if you have more than 1 ACTIVE nitrate instance?  Which of the different dust instances would map into the different nitrate instances?  Since I do not anticipate needing more than 1 ACTIVE nitrate instance, I’d suggest that you should test for it and throw an exception if you have more than 1 ACTIVE nitrate instance (“more than 1 ACTIVE nitrate instance not implemented yet.”). If we ever need to support that, then the solution is to have this coupling specified via rc file. For now, a single ACTIVE nitrate component is all that we need. And all PASSIVE components are fed the same du/ss/so4 active instances.
 
   Does it make sense?
 
          Arlindo
"

**Comment by mathomp4** [comment]
@atrayano Has this been done?

**Comment by mathomp4** [comment]
Reexamined on 2021-Apr-12 with @bena-nasa and @tclune 

@atrayano Is this done?

**Comment by mathomp4** [comment]
@tclune @atrayano I don't think this has been resolved, correct? But it seems like something that is needed, so I'll add the "long term" to this.

**Comment by mathomp4** [comment]
Pinging @tclune

**Comment by tclune** [comment]
I see now that @atrayano has never commented on this ticket.   Have to lean on him to at least assess the difficulty.   I can then check with @amdasilva on priority.   But I think he mentioned something related to this quite recently.
